### PR TITLE
proposed solution for addpath(genpath(...)) problems

### PR DESCRIPTION
**Pull request by robertoostenveld**
as brought up by @gllmflndn in issue #899

### CODE REVIEW

**Comment by robertoostenveld** [comment]
there are 16 functions 
```
mac011> ls matlablt201*/*.m | wc -l
16
```

that should get this piece of code. Partially it is identical for all, but the actual function call (with input and output args) would have to be customized. Still quite doable.

**Comment by robertoostenveld** [comment]
these 5 would have to be dealt with differently 

```
>> fn = {
'iscolumn'
'isrow'
'ismatrix'
'narginchk'
'isequaln'
'istable'
'webread'
'weboptions'
'webwrite'
'endsWith'
'contains'
'startsWith'
'newline'
'pad'
'isfolder'
'isfile'
};

for i=1:numel(fn)
  if ~exist(fn{i}, 'builtin')
    disp(fn{i});
  end
end

istable
webread
weboptions
webwrite
pad
```

although ... I also realize that Mathworks is sometimes (over versions) changing a function into a builtin.  So it should not only test for builtins.

File: compat/matlablt2016b/contains.m [hunk 1]
```diff
@@ -7,6 +7,33 @@
 % This is a compatibility function that should only be added to the path on
 % MATLAB versions prior to 2016b.
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% see https://github.com/fieldtrip/fieldtrip/issues/899
+
+if exist(mfilename, 'builtin') || strncmp(mfilename('fullpath'), matlabroot, length(matlabroot))
```
Comment (robertoostenveld) [inline compat/matlablt2016b/contains.m]: the 2nd part does not work like this

**Comment by robertoostenveld** [comment]
how do you like this @gllmflndn?

**Comment by gllmflndn** [comment]
Thanks @robertoostenveld for looking into this.

I think your changes would help minimise disruptions. When testing using the (unusual) approach described in #899:

> For example, using the MATLAB interface, just changing directory to `fieldtrip/compat/matlablt2016b/` and double-clicking on a file to open it will fail with error `PATTERN must be a string scalar or character vector`.

I notice a number of things:

- `ft_warning` is not in the path so an error still appears. Maybe the compatibility functions should not call any `ft_` functions?
- `rmpath` displays a warning that the directory to be removed is not in the path so there could be a check that it is explicitly in the path first, teamed with `~isdeployed`.

I would agree this is a peculiar corner case but it would be great to have a solution as robust as possible. There will anyway be a number of warnings being displayed:

> Warning: Function xxxx has the same name as a MATLAB builtin. We suggest you rename the function to avoid a potential name conflict.

**Comment by robertoostenveld** [comment]
thanks for the good suggestions, I will follow that up and also implement it for the remaining 14 functions.

**Pull request merged**